core/src/types/list.rs

```diff
@@ -80,44 +83,81 @@ impl Stream for Lister {
             return Poll::Ready(None);
         }
 
-        if let Some(fut) = self.stating.as_mut() {
-            let (path, rp) = ready!(fut.poll_unpin(cx));
-
-            // Make sure we will not poll this future again.
-            self.stating = None;
-            let metadata = match rp {
-                Ok(rp) => rp.into_metadata(),
+        while self.task_queue.len() < self.task_queue.capacity() {
+            match ready!(self.lister.poll_next(cx)) {
```

Comment:
we can go to poll stat if `poll_next` returns Pending.

Reply:
I'm confused, doesn't that result in only one join handle being created as we discussed earlier?

Reply:
> I'm confused, doesn't that result in only one join handle being created as we discussed earlier?

The trick is to perform work while `poll_next` returns `Pending` instead of returning directly. Loop here will add the lantency for `Lister` to return entries, users will need to fill the whole taks queue until they got their first entry.

Reply:
I almost forgot that we need to keep the entry order, so we must wait for the latest entry to finish its stat before we can process the rest. Now I can understand why you want poll stat if `poll_next` returns Pending.